### ISSUE TITLE
[Bug] Transport rate limits from config not work

### ISSUE THREAD

**Issue by ExPS-Pisko** [thingsboard/thingsboard#3556]
Thingsboard CE 3.0

I use TB Gateway 2.3.1 (Python) + TB 3.0. I have configured a device with 20 telemetry Tags (values).

I experienced a limit in TB, that max. 10 variables are updated when the Telemetry sampling is 1000ms (i.e. sampling 20 values per second). Only 10 of these 20 values get updated. I tried to configure thingsboard.yml in this way:

transport:
  sessions:
    inactivity_timeout: "${TB_TRANSPORT_SESSIONS_INACTIVITY_TIMEOUT:300000}"
    report_timeout: "${TB_TRANSPORT_SESSIONS_REPORT_TIMEOUT:30000}"
  rate_limits:
    enabled: "${TB_TRANSPORT_RATE_LIMITS_ENABLED:true}"
    tenant: "${TB_TRANSPORT_RATE_LIMITS_TENANT:1000:1,20000:60}"
    device: "${TB_TRANSPORT_RATE_LIMITS_DEVICE:100:1,3000:60}"

But it has no effect (see the Devices transport rate limit increased from 10:1 to 100:1

Best regards Michal Pisko

**Comment by zbeacon** [comment]
Hi @ExPS-Pisko 

Thank you for your feedback, but it is not a bug. Limits in thingsboard.yaml works for messages arrived by transport, but the gateway pack messages from device into one message to ThingsBoard to decrease count of messages and work with network. For example:

We received following messages from device using MQTT connector:

{"deviceName":"SomeDevice","model":"AIKB23","temp":23}
{"deviceName":"SomeDevice","model":"AIKB23","temp":24}
{"deviceName":"SomeDevice","model":"AIKB23","temp":23}
{"deviceName":"SomeDevice","model":"AIKB23","temp":25}
{"deviceName":"SomeDevice","model":"AIKB23","temp":26}
{"deviceName":"SomeDevice","model":"AIKB23","temp":27}

The gateway will process these messages into one, like the following one and send it to ThingsBoard.

```
{
  "deviceName":"SomeDevice", 
  "attributes":[{"model":"AIKB23"}], 
  "telemetry":[
    {"ts":351351351,"temp":23},
    {"ts":351351345,"temp":24},
    {"ts":351351398,"temp":23},
    {"ts":351351342,"temp":25},
    {"ts":353523452,"temp":26},
    {"ts":355834235,"temp":27}
]}

So It will be the only one message fro ThingsBoard. 
To set count of messages in message to ThingsBoard you can set up value read_records_count or max_read_records_count in your tb_gateway.yaml, it will set the max size of messages. 

You can read more [here](https://thingsboard.io/docs/iot-gateway/configuration/?storageConfig=memory#storage-configuration).
```

**Comment by ExPS-Pisko** [comment]
Dear Mr. Barkov,

 

Thank you for your feedback!

 

However, I am still not able to get a desired performance, regardless of the tb_gateway.yaml setup.

 

Attached both configuration files.

 

I changed to disabled subscription, forcing a periodic read-out of data from OPC UA server. When the sampling time is 3000 ms, all 20 data points (telemetry data) are read from OPC UA server and stored correctly in TB. However, when I set 1000 ms sampling period (Scan Period), I get only 10 from these 20 values (i.e. exactly 10 values per second – see attached screenshot, OPC1 to OPC10 are empty, also directly at the device, these 10 values do not change + the timestamps do not change).

 

Which parameters shall I change in order to get all 20 telemetry data for 1 device with a period of 1 second? read_records_count or max_read_records_count have no impact on the above described performance…

 

Thanks for help and best regards

Michal Pisko
